daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala

```diff
@@ -999,6 +1000,60 @@ class DecodeV1Spec
       }
     }
 
+    s"Reject interface definitions in modules iff version < ${LV.Features.interfaces}" in {
+      val i = DamlLf1.DefInterface.newBuilder()
+      i.setTyconInternedDname(1)
+      i.setParamInternedStr(0)
+      i.setPrecond(unitExpr)
```

Comment:
Why using two different styles for building proto messages (in line 1013-10017 you use the fact that  `.setX`  nethod return the builder). 

```suggestion
      val i = DamlLf1.DefInterface.newBuilder()
        .setTyconInternedDname(1)
        .setParamInternedStr(0)
        .setPrecond(unitExpr)
        .build()
```

furthermore the final `build()`, ensure you get an immutable object instead of a mutable builder.